### DISCUSSION TITLE
How to define a pod to not have any inbound ports without an error in the log?

### DISCUSSION THREAD

**Discussion by karlparry** [linkerd/linkerd2#7765] [Q&A] [answered]
We have a few pods that have no inbound listening ports. When these run we get this error in the Proxy log. 

Is there anyway to stop these messages appearing in the Proxy logs?

```log
ERROR ThreadId(01) linkerd_app::env: No inbound ports specified via LINKERD2_PROXY_INBOUND_PORTS
```

Using Linkerd version 2.11.X

**Comment by mateiidavid** [comment, marked as answer]
Hey @karlparry, we use the env variable to discover authorization policies (see: [implementation ticket](https://github.com/linkerd/linkerd2/issues/6267)). We discussed the usage of `containerPorts` in https://github.com/linkerd/linkerd2/issues/7640.

Anyway, looking through the code, it seems that when the env variable isn't set, we'll [log out a warn message](https://github.com/linkerd/linkerd2-proxy/blob/bb26cdc8c9a96161c493ae9b467370dc8c47431e/linkerd/app/src/env.rs#L535-L544) and use a default policy. Further in the code, when we try to resolve the  address for the policy controller, we seem to check the ports again and log an error message if they're missing. AFAIK based on this, the error shouldn't have a big impact. Might be some room here to turn it into a warn, or something similar, but I'm not (_personally_) sure of what the implications would be.

Hope this helps? Let me know what you think.

**Reply by karlparry** [reply]
Thanks for the response. That is exactly what I saw when I read the code, but didn't want to assume. 
You are right, the error has no impact, and only appears once on start up so it isn't a problem at all.

**Reply by olix0r** [reply]
@mateiidavid is correct. It would probably be best if the injector always set the environment variable, even if empty, so that the proxy wouldn't log an error. The proxy's log message is mostly intended to indicate that an expect configuration wasn't set (i.e. if an older injector is trying to configure a newer proxy).

**Comment by DrFaust92** [comment]
Im encountering this as well, is it worth to open a feature request/submit a PR to change this to WARN instead of error?
